types/ember-data/test/transform.ts

```diff
@@ -15,8 +16,41 @@ class PointTransform extends DS.Transform {
     }
 }
 
+class EnumTransform extends DS.Transform {
+    deserialize<AllowedValue>(
+        serialized: any,
+        options: { allowedValues: Record<string, AllowedValue> },
+    ): AllowedValue {
+        const { allowedValues } = options;
+        const allowedValuesArr = Object.keys(allowedValues).map((key) => allowedValues[key]);
+
+        assert(
+            `Value "${serialized}" must be one of "${allowedValuesArr.join(', ')}"`,
+            allowedValuesArr.indexOf(serialized) >= 0,
+        );
+
+        return serialized;
+    }
+
+    serialize<AllowedValue>(
+        deserialized: AllowedValue,
+        options: { allowedValues: Record<string, AllowedValue> },
```

Comment:
Here, since there's no use of the type other than in the `options`, it doesn't even need to be present as a generic.

```suggestion
    serialize(
        deserialized: AllowedValue,
        options: { allowedValues: Record<string, unknown> },
```